crates/wasmtime/src/runtime/vm/gc/enabled/free_list.rs

```diff
@@ -47,21 +121,11 @@ impl FreeList {
     fn check_layout(&self, layout: Layout) -> Result<u32> {
         ensure!(
             layout.align() <= ALIGN_USIZE,
-            "requested allocation's alignment of {} is greater than max supported alignment of {ALIGN_USIZE}",
+            "requested allocation's alignment of {} is greater than max supported \
+             alignment of {ALIGN_USIZE}",
             layout.align(),
         );
 
-        if layout.size() > self.max_size() {
-            let trap = crate::Trap::AllocationTooLarge;
-            let err = anyhow::Error::from(trap);
-            let err = err.context(format!(
-                "requested allocation's size of {} is greater than the max supported size of {}",
-                layout.size(),
-                self.max_size(),
-            ));
-            return Err(err);
-        }
```

Comment:
To confirm, there's still a test below testing this behavior and that it fails?

Reply:
Correct, we just return an `Ok(None)` (meaning that the allocation might succeed after growth/gc) instead of `Err(_)` (meaning that the allocation will never succeed no matter what)